storage/innobase/handler/ha_innodb.cc

```diff
@@ -16003,6 +16069,9 @@ static int innodb_drop_tablespace(handlerton *hton, THD *thd,
   trx_t *trx = check_trx_exists(thd);
   TrxInInnoDB trx_in_innodb(trx);
   trx_start_if_not_started(trx, true);
+  if (!innobase_is_ddse()) {
```

Comment:
for competence, do we also need to do for alter_tabespaces? or all cmd in innobase_alter_tablespace()

Reply:
Apparently not, they get registered with the server 2PC. What some of them do need though, is marking them as RW transactions